### PR TITLE
JIRA: VZ-7172 mysql values webhook to check for server podSpec modification

### PR DESCRIPTION
**Pull request by jmaron99**
A webhook to check whether a server podSpec has been specified and, if so, warn that pod spec modifications do not automatically trigger a stateful set restart.

warning for podSpec modification:
`
Warning: Modifications to MySQL server pod specs do not trigger an automatic restart of the stateful set. Please refer to the documentation for a rolling restart: https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#rollout
verrazzano.install.verrazzano.io/my-verrazzano configured
`

### CODE REVIEW

**Review by mgianatagh — DISMISSED** [review]

File: platform-operator/apis/verrazzano/v1alpha1/mysql_values_webhook.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			return validateMysqlValues(vz)
+		}
+	}
+	return admission.Allowed("")
+}
+
+// validateMysqlValues presents the user with a warning if there are podSpecs specified as overrides.
+func validateMysqlValues(vz *Verrazzano) admission.Response {
+	response := admission.Allowed("")
+	if isMinVersion(vz.Spec.Version, MIN_VERSION) {
```
Comment (pfmackin) [inline platform-operator/apis/verrazzano/v1alpha1/mysql_values_webhook.go:58]: even though this works, better happy path go style is.  Don't think you need to change it.

if !isMinVersion(..) {
  return ..
}

if vz.Spec.Components.Keycloak == nil {
  return .,..
}

var overrrides...

File: platform-operator/apis/verrazzano/v1beta1/mysql_values_webhook.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
```
Comment (pfmackin) [inline platform-operator/apis/verrazzano/v1beta1/mysql_values_webhook.go:1]: Is there common code between alpha and beta that can be shared?

**Review by pfmackin — DISMISSED** [review]

File: platform-operator/apis/verrazzano/webhooks/mysql_values_webhook_v1alpha1.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	decoder *admission.Decoder
+}
+
+// InjectDecoder injects the decoder.
+func (v *MysqlValuesValidatorV1alpha1) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// Handle performs validation of created or updated Verrazzano resources.
```
Comment (pfmackin) [inline platform-operator/apis/verrazzano/webhooks/mysql_values_webhook_v1alpha1.go:28]: duplicate comment

File: platform-operator/apis/verrazzano/webhooks/mysql_values_webhook_v1alpha1.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+// Handle performs validation of created or updated Verrazzano resources.
+func (v *MysqlValuesValidatorV1alpha1) Handle(ctx context.Context, req admission.Request) admission.Response {
+
```
Comment (pfmackin) [inline platform-operator/apis/verrazzano/webhooks/mysql_values_webhook_v1alpha1.go:31]: normally do not have blank line after function signature

File: platform-operator/apis/verrazzano/webhooks/mysql_values_webhook_v1beta1.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+// InjectDecoder injects the decoder.
+func (v *MysqlValuesValidatorV1beta1) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// Handle performs validation of created or updated Verrazzano resources.
+func (v *MysqlValuesValidatorV1beta1) Handle(ctx context.Context, req admission.Request) admission.Response {
+
```
Comment (pfmackin) [inline platform-operator/apis/verrazzano/webhooks/mysql_values_webhook_v1beta1.go:34]: normally do not have blank line after function signature

**Review by stevshap — APPROVED** [review]

**Review by stevshap — APPROVED** [review]

**Review by pfmackin — DISMISSED** [review]

**Review by stevshap — DISMISSED** [review]

**Review by mgianatagh — APPROVED** [review]

**Review by stevshap — APPROVED** [review]

**Pull request merged**